pykube/http.py

```diff
@@ -71,17 +80,96 @@ def _auth_gcp(self, request, token, expiry, config):
         )
 
         if should_persist and config:
-            self._persist_credentials(config, credentials.token, credentials.expiry)
+            auth_opts = {
+                "access-token": credentials.token,
+                "expiry": credentials.expiry,
+            }
+            self._persist_credentials(config, auth_opts)
 
         def retry(send_kwargs):
             credentials.refresh(auth_request)
             response = self.send(original_request, **send_kwargs)
             if response.ok and config:
-                self._persist_credentials(config, credentials.token, credentials.expiry)
+                auth_opts = {
+                    "access-token": credentials.token,
+                    "expiry": credentials.expiry,
+                }
+                self._persist_credentials(config, auth_opts)
             return response
 
         return retry
 
+    def _is_valid_jwt(self, token):
+        """Validate JWT token for correctness and near expiration"""
+        if not token:
+            return False
+        reserved_characters = frozenset(["=", "+", "/"])
+        if any(char in token for char in reserved_characters):
+            # Invalid jwt, as it contains url-unsafe chars
+            return False
+        parts = token.split(".")
+        if len(parts) != 3:  # Not a valid JWT
+            return False
+        padding = (4 - len(parts[1]) % 4) * "="
+        if len(padding) == 3:
+            # According to spec, 3 padding characters cannot occur
+            # in a valid jwt
+            # https://tools.ietf.org/html/rfc7515#appendix-C
+            return False
+        jwt_attributes = json.loads(
+            base64.b64decode(parts[1] + padding).decode("utf-8")
+        )
+        expire = jwt_attributes.get("exp")
+        # allow missing exp, but deny tokens that are about to expire soon
+        return expire is None or (
+            datetime.datetime.fromtimestamp(expire, tz=UTC)
+            - EXPIRY_SKEW_PREVENTION_DELAY
+        ) > datetime.datetime.utcnow().replace(tzinfo=UTC)
+
+    def _refresh_oidc_token(self, config):
+        if not oidc_auth_installed:
+            raise ImportError(
+                "missing dependencies for OIDC token refresh support "
+                "(try pip install pykube-ng[oidc]"
+            )
+        auth_config = config.user["auth-provider"]["config"]
+        if "idp-certificate-authority" in auth_config:
+            verify = auth_config["idp-certificate-authority"].filename()
+        else:
+            verify = None
+        # TODO add timeouts
+        response = requests.get(
+            f"{auth_config['idp-issuer-url']}/.well-known/openid-configuration",
+            verify=verify,
+        )
+
+        if response.status_code != HTTPStatus.OK:
```

Comment:
if you want to catch all errors (e.g. connection problems), you would need to wrap it in `try`/`except` --- I assume you want to ignore failures and continue with the old token